### PR TITLE
unstoppableswap: init at 0.6.3

### PR DESCRIPTION
**Pull request by linsui**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/un/unstoppableswap/package.nix [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    desktopName = "UnstoppableSwap";
+    comment = "A modern GUI video and audio downloader";
+    categories = [ "Utility" ];
+    startupWMClass = "UnstoppableSwap";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  postInstall = ''
+    makeWrapper ${electron}/bin/electron $out/bin/UnstoppableSwap \
```
Comment (SuperSandro2000) [inline pkgs/by-name/un/unstoppableswap/package.nix:59]:
```suggestion
    makeWrapper ${lib.getExe electron} $out/bin/UnstoppableSwap \
```

File: pkgs/by-name/un/unstoppableswap/package.nix [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        --add-flags $out/lib/node_modules/unstoppableswap-gui/dist/main/main.js
+
+    mkdir -p $out/lib/node_modules/build/bin
+    ln -s ${lib.getBin tor}/bin/tor $out/lib/node_modules/build/bin/tor
+    ln -s ${lib.getBin xmr-btc-swap}/bin/swap $out/lib/node_modules/build/bin/swap
+
+    install -Dm444 ../../assets/icons/512x512.png $out/share/icons/hicolor/512x512/apps/UnstoppableSwap.png
+    install -Dm444 "${desktopItem}/share/applications/"* -t $out/share/applications
```
Comment (SuperSandro2000) [inline pkgs/by-name/un/unstoppableswap/package.nix:67]: Please use copyDesktopItems instead

File: pkgs/by-name/xm/xmr-btc-swap/package.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+, protobuf
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "xmr-btc-swap";
+  version = "0.13.1";
+
+  src = fetchFromGitHub {
+    owner = "comit-network";
+    repo = pname;
```
Comment (SuperSandro2000) [inline pkgs/by-name/xm/xmr-btc-swap/package.nix:13]:
```suggestion
    repo = "xmr-btc-swap";
```

File: pkgs/by-name/xm/xmr-btc-swap/package.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  };
+
+  postPatch = ''
+    rm swap/build.rs
+  '';
+
+  nativeBuildInputs = [ protobuf ];
+
+  env.VERGEN_GIT_DESCRIBE = version;
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/by-name/xm/xmr-btc-swap/package.nix:34]: This needs a very short comment on why the tests are disabled

Comment (linsui) [reply pkgs/by-name/xm/xmr-btc-swap/package.nix]: I don't really know why the test failed. It requires docker and fails after a long time with "failed to create docker network".

File: pkgs/by-name/un/unstoppableswap/package.nix [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+    install -Dm444 ../../assets/icons/512x512.png $out/share/icons/hicolor/512x512/apps/UnstoppableSwap.png
+  '';
+
+  meta = {
+    description = "XMR<>BTC Atomic Swaps GUI";
+    homepage = "https://github.com/UnstoppableSwap/unstoppableswap-gui";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ linsui ];
+    platforms = lib.platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/by-name/un/unstoppableswap/package.nix:75]:
```suggestion
    platforms = [ "x86_64-linux" ];
```

All other platforms fail to build. I tested aarch64-linux:

```
xmr-btc-swap-aarch64-linux> error[E0425]: cannot find value `DOWNLOAD_URL` in this scope
xmr-btc-swap-aarch64-linux>    --> swap/src/monero/wallet_rpc.rs:228:41
xmr-btc-swap-aarch64-linux>     |
xmr-btc-swap-aarch64-linux> 228 |             let response = reqwest::get(DOWNLOAD_URL).await?;
xmr-btc-swap-aarch64-linux>     |                                         ^^^^^^^^^^^^ not found in this scope
xmr-btc-swap-aarch64-linux>
xmr-btc-swap-aarch64-linux> error[E0425]: cannot find value `DOWNLOAD_URL` in this scope
xmr-btc-swap-aarch64-linux>    --> swap/src/monero/wallet_rpc.rs:238:30
xmr-btc-swap-aarch64-linux>     |
xmr-btc-swap-aarch64-linux> 238 |                 download_url=DOWNLOAD_URL,
xmr-btc-swap-aarch64-linux>     |                              ^^^^^^^^^^^^ not found in this scope
xmr-btc-swap-aarch64-linux>
xmr-btc-swap-aarch64-linux> error[E0425]: cannot find value `DOWNLOAD_URL` in this scope
xmr-btc-swap-aarch64-linux>    --> swap/src/monero/wallet_rpc.rs:275:38
xmr-btc-swap-aarch64-linux>     |
xmr-btc-swap-aarch64-linux> 275 |                         download_url=DOWNLOAD_URL,
xmr-btc-swap-aarch64-linux>     |                                      ^^^^^^^^^^^^ not found in this scope
xmr-btc-swap-aarch64-linux>
xmr-btc-swap-aarch64-linux> error[E0425]: cannot find value `DOWNLOAD_URL` in this scope
xmr-btc-swap-aarch64-linux>    --> swap/src/monero/wallet_rpc.rs:286:30
xmr-btc-swap-aarch64-linux>     |
xmr-btc-swap-aarch64-linux> 286 |                 download_url=DOWNLOAD_URL,
xmr-btc-swap-aarch64-linux>     |                              ^^^^^^^^^^^^ not found in this scope
xmr-btc-swap-aarch64-linux>
xmr-btc-swap-aarch64-linux> error[E0425]: cannot find value `DOWNLOAD_HASH` in this scope
xmr-btc-swap-aarch64-linux>    --> swap/src/monero/wallet_rpc.rs:292:31
xmr-btc-swap-aarch64-linux>     |
xmr-btc-swap-aarch64-linux> 292 |             if result_hash != DOWNLOAD_HASH {
xmr-btc-swap-aarch64-linux>     |                               ^^^^^^^^^^^^^ not found in this scope
xmr-btc-swap-aarch64-linux>
xmr-btc-swap-aarch64-linux> error[E0425]: cannot find value `DOWNLOAD_HASH` in this scope
xmr-btc-swap-aarch64-linux>    --> swap/src/monero/wallet_rpc.rs:296:21
xmr-btc-swap-aarch64-linux>     |
xmr-btc-swap-aarch64-linux> 296 |                     DOWNLOAD_HASH
xmr-btc-swap-aarch64-linux>     |                     ^^^^^^^^^^^^^ not found in this scope
xmr-btc-swap-aarch64-linux>
xmr-btc-swap-aarch64-linux> warning: unused import: `big_bytes::BigByte`
xmr-btc-swap-aarch64-linux>  --> swap/src/monero/wallet_rpc.rs:3:5
xmr-btc-swap-aarch64-linux>   |
xmr-btc-swap-aarch64-linux> 3 | use big_bytes::BigByte;
xmr-btc-swap-aarch64-linux>   |     ^^^^^^^^^^^^^^^^^^
xmr-btc-swap-aarch64-linux>   |
xmr-btc-swap-aarch64-linux>   = note: `#[warn(unused_imports)]` on by default
xmr-btc-swap-aarch64-linux>
xmr-btc-swap-aarch64-linux> warning: unused import: `TryStreamExt`
xmr-btc-swap-aarch64-linux>  --> swap/src/monero/wallet_rpc.rs:5:26
xmr-btc-swap-aarch64-linux>   |
xmr-btc-swap-aarch64-linux> 5 | use futures::{StreamExt, TryStreamExt};
xmr-btc-swap-aarch64-linux>   |                          ^^^^^^^^^^^^
xmr-btc-swap-aarch64-linux>
xmr-btc-swap-aarch64-linux> For more information about this error, try `rustc --explain E0425`.
xmr-btc-swap-aarch64-linux> warning: `swap` (lib) generated 2 warnings
xmr-btc-swap-aarch64-linux> error: could not compile `swap` (lib) due to 6 previous errors; 2 warnings emitted
```

File: pkgs/by-name/xm/xmr-btc-swap/package.nix [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  nativeBuildInputs = [ protobuf ];
+
+  env.VERGEN_GIT_DESCRIBE = version;
+
+  doCheck = false; # The test requires docker and network
+
+  meta = with lib; {
+    description = "Bitcoin–Monero Cross-chain Atomic Swap";
+    homepage = "https://github.com/comit-network/xmr-btc-swap";
+    license = with licenses; [ gpl3Only ];
```
Comment (SuperSandro2000) [inline pkgs/by-name/xm/xmr-btc-swap/package.nix:41]:
```suggestion
    license = with licenses; [ gpl3Only ];
    platforms = [ "x86_64-linux" ];
```

File: pkgs/by-name/un/unstoppableswap/package.nix [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+    copyDesktopItems
+  ];
+
+  postInstall = ''
+    makeWrapper ${lib.getExe electron} $out/bin/UnstoppableSwap \
+        --add-flags $out/lib/node_modules/unstoppableswap-gui/dist/main/main.js
```
Comment (SuperSandro2000) [inline pkgs/by-name/un/unstoppableswap/package.nix:65]:
```suggestion
      --add-flags $out/lib/node_modules/unstoppableswap-gui/dist/main/main.js
```

**Comment by linsui** [comment]
I tried to update unstoppableswap to 0.6.1 but failed.

**Comment by linsui** [comment]
OK, it's just network issue...

**Comment by linsui** [comment]
These package is still not mature enough to be usable so I'm closing this PR. Sorry about that.

**Pull request closed without merging**